subprojects/core/src/test/groovy/org/gradle/internal/progress/TaskOutcomeStatisticsFormatterTest.groovy

```diff
@@ -36,28 +36,58 @@ class TaskOutcomeStatisticsFormatterTest extends Specification {
         expect:
         formatter.incrementAndGetProgress(taskState(UP_TO_DATE)) == " [100% AVOIDED, 0% DONE]"
         formatter.incrementAndGetProgress(taskState(FROM_CACHE)) == " [100% AVOIDED, 0% DONE]"
-        formatter.incrementAndGetProgress(taskState(NO_SOURCE)) == " [100% AVOIDED, 0% DONE]"
-        formatter.incrementAndGetProgress(taskState(SKIPPED)) == " [100% AVOIDED, 0% DONE]"
+    }
+
+    def "tasks with NO-SOURCE are not counted"() {
+        given:
+        formatter.incrementAndGetProgress(taskState(EXECUTED))
+
+        expect:
+        formatter.incrementAndGetProgress(taskState(NO_SOURCE)) == " [0% AVOIDED, 100% DONE]"
+    }
+
+    def "SKIPPED tasks are not counted"() {
+        given:
+        formatter.incrementAndGetProgress(taskState(EXECUTED))
+
+        expect:
+        formatter.incrementAndGetProgress(taskState(SKIPPED)) == " [0% AVOIDED, 100% DONE]"
+    }
+
+    def "tasks with no actions are not counted"() {
+        given:
+        formatter.incrementAndGetProgress(taskState(UP_TO_DATE))
+
+        expect:
+        formatter.incrementAndGetProgress(taskState(EXECUTED, false)) == " [100% AVOIDED, 0% DONE]"
```

Comment:
do you also need an `taskState(UP_TO_DATE, false)` test too (an up-to-date lifecycle task)?